### PR TITLE
MEN-1355: mender-artifact now fails with whitespace in the artifact-name

### PR DESCRIPTION
**Pull request by oleorhagen**
Changelog: Title

Signed-off-by: Ole Petter <ole.orhagen@cfengine.com>

### CODE REVIEW

**Comment by kacf** [comment]
Alright, QEMU test passed. It looks like Travis is failing because it measures the [cyclomatic complexity](https://en.wikipedia.org/wiki/Cyclomatic_complexity) is too high. A bit unfair in this case, since your change is tiny. Probably the function was very close to the limit already. Maybe you can get around this by adding the check to the parameter validation instead of during writing?

File: artifacts.go [hunk 1]
```diff
@@ -523,8 +523,13 @@ func run() error {
...
 	writeRootfs := cli.Command{
-		Name:   "rootfs-image",
-		Action: writeArtifact,
+		Name: "rootfs-image",
+		Action: func(c *cli.Context) error {
+			if len(strings.Fields(c.String("artifact-name"))) > 1 { // check for whitespace in artifact-name
+				return cli.NewExitError("whitespace is not allowed in the artifact-name", 1)
+			}
+			return writeArtifact(c)
+		},
```
Comment (oleorhagen) [inline artifacts.go]: @kacf took a slightly different approach when rewriting this, and decided to wrap the writeartifact function. Not super clean, but acceptable?

Comment (kacf) [reply artifacts.go]: No, I think this is quite clean actually, as long as it's as small as that!

**Comment by kacf** [comment]
@oleorhagen, @GregorioDiStefano: This failed twice with seemingly the same error:

* https://mender-jenkins.mender.io/job/mender-builder/2108/console
* https://mender-jenkins.mender.io/job/mender-builder/2114/console

Greg, the tests are otherwise in a good state, right? Could it be a real error in the PR?

**Pull request merged**